test/extensions/transport_sockets/tls/cert_validator/default_validator_test.cc

```diff
@@ -116,6 +116,34 @@ TEST(DefaultCertValidatorTest, TestMatchSubjectAltNameNotMatched) {
   EXPECT_FALSE(DefaultCertValidator::matchSubjectAltName(cert.get(), subject_alt_name_matchers));
 }
 
+TEST(DefaultCertValidatorTest, TestCertificateVerificationWithSANMatcher) {
+  Stats::TestUtil::TestStore test_store;
+  SslStats stats = generateSslStats(test_store);
+  // Create the default validator object.
+  auto default_validator =
+      std::make_unique<Extensions::TransportSockets::Tls::DefaultCertValidator>(
+          /*CertificateValidationContextConfig=*/nullptr, stats,
+          Event::GlobalTimeSystem().timeSystem());
+
+  bssl::UniquePtr<X509> cert = readCertFromFile(TestEnvironment::substitute(
+      "{{ test_rundir }}/test/extensions/transport_sockets/tls/test_data/san_dns_cert.pem"));
+  envoy::type::matcher::v3::StringMatcher matcher;
+  matcher.MergeFrom(TestUtility::createRegexMatcher(".*.example.com"));
+  std::vector<Matchers::StringMatcherImpl> san_matchers;
+  san_matchers.push_back(Matchers::StringMatcherImpl(matcher));
+  // Verify the certificate with correct SAN regex matcher.
+  EXPECT_EQ(default_validator->verifyCertificate(cert.get(), /*verify_san_list=*/{}, san_matchers),
+            Envoy::Ssl::ClientValidationStatus::Validated);
```

Comment:
Also correctness of `stats` could be checked here: no increment in case of a matching SAN and incremented otherwise.

Reply:
Thanks for review!   I explored a few options for testing `stats`:
1. test_server_->counter(....). The problem is this test is free function test (i.e. no test fixture) so it doesn't have necessary infra setup. I could try to add it though but I feel it is a bit overkill here.
2. Exposing `stats_` in `DefaultCertValidator` class. But I feel it is not ideal to do this just for testing purpose.

I am new to Envoy codebase and I am wondering if you have other suggested ways to check `stats` easily? Thanks

Reply:
No need to expose anything. `stats` are created in the line 121 and accessible farther down the test. Just put something like
```c++
EXPECT_EQ(0, stats.fail_verify_san_.value());
```
where SAN is expected to be valid and
```c++
EXPECT_EQ(1, stats.fail_verify_san_.value());
```
otherwise.

Reply:
Done. thanks! I didn't notice that it is pass-by-ref and reference class member. I guess this is envoy's intended dependency pattern and disallow copying (i.e. copy assignment) and moving. There should be no lifetime concern here.